modules/light-clients/07-tendermint/misbehaviour_handle.go

```diff
@@ -53,6 +53,29 @@ func (cs ClientState) CheckForMisbehaviour(ctx sdk.Context, cdc codec.BinaryCode
 	case *Misbehaviour:
 		// The correctness of Misbehaviour ClientMessage types is ensured by calling VerifyClientMessage prior to this function
 		// Thus, here we can return true, as ClientMessage is of type Misbehaviour
+		// if heights are equal check that this is valid misbehaviour of a fork
+		// otherwise if heights are unequal check that this is valid misbehavior of BFT time violation
+		if msg.Header1.GetHeight().EQ(msg.Header2.GetHeight()) {
+			blockID1, err := tmtypes.BlockIDFromProto(&msg.Header1.SignedHeader.Commit.BlockID)
+			if err != nil {
+				return true
+			}
+
+			blockID2, err := tmtypes.BlockIDFromProto(&msg.Header2.SignedHeader.Commit.BlockID)
+			if err != nil {
+				return true
+			}
```

Comment:
Given the method signature only returns a `bool` I think it would be more appropriate to return false in both of these conditionals. In reality, `BlockIDFromProto` should never return errors here as its checked in [`ValidateBasic`](https://github.com/cosmos/ibc-go/blob/main/modules/light-clients/07-tendermint/misbehaviour.go#L91-L98) before this code would be reached.

The alternative option is to `panic` given its unreachable code, but I think I would rather opt for returning false.